nodes/core/io/21-httprequest.js

```diff
@@ -135,7 +135,7 @@ module.exports = function(RED) {
             }
             var payload = null;
 
-            if (typeof msg.payload !== "undefined" && (method == "POST" || method == "PUT" || method == "PATCH" ) ) {
+            if (msg.payload && (method == "POST" || method == "PUT" || method == "PATCH" || method == "PROPPATCH" || method == "PROPFIND" ) ) {
```

Comment:
This will prevent any payload that has a falsely value being allowed through, such as:
 - `0`
 - `false`

That is why the code does an explicit check for `undefined`.